runtime/testnet/src/extensions.rs

```diff
@@ -0,0 +1,740 @@
+use cumulus_pallet_parachain_system::RelaychainDataProvider;
```

Comment:
I think we need to refactor the chain extension into common in a separate PR if possible.

Reply:
Agree